### ISSUE TITLE
Mapped types do not respect @public jsdoc

### ISSUE THREAD

**Issue by chrisd08** [timocov/ts-transformer-properties-rename#15]
Mapped types do not appear to respect @public jsdoc

```ts
/** @public */
export type TestKey = "a" | "b";
/** @public */
export type TestMappedType = {
  /** @public */
  [P in TestKey]?: {};
};

class TestMappedClass {
  /** @public */
  private testMapped: TestMappedType = {};
  public constructor() {
    this.testMapped.a = {};
  }
}
```

Results in

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TestMappedClass = /** @class */ (function () {
    function TestMappedClass() {
        /** @public */
        this._private_testMapped = {};
        this._private_testMapped._internal_a = {};
    }
    return TestMappedClass;
}());
```

Note: I tried it using Record as well and it has the same issue

**Comment by timocov** [comment]
@chrisd08 expected output here is:

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TestMappedClass = /** @class */ (function () {
    function TestMappedClass() {
        this._private_testMapped = {};
        this._private_testMapped.a = {};
    }
    return TestMappedClass;
}());
```

am I right (don't rename `a`)?

**Comment by chrisd08** [comment]
@timocov Yes that's correct

**Comment by timocov** [comment]
Fixed in 74ed470. If a mapped type is exported, you don't need to mark it as public now. But anyway, if a mapped type isn't exported, you can mark the whole type as public in this case (instead of indexer or property keys).

**Comment by timocov** [comment]
The fix is just published in [0.8.0](https://github.com/timocov/ts-transformer-properties-rename/releases/tag/v0.8.0) version. Let me know if you have any further issues.
